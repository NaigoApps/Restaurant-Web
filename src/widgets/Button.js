import React, {Component} from 'react';

/**
 * Expects:
 * - name: label caption
 * - changeAction: action to throw on change
 */

export default class Button extends Component {
    constructor(props) {
        super(props);
    }

    clickAction() {
        if (this.props.commitAction) {
            this.props.commitAction();
        }
    }

    getClassName() {
        let classes = ["btn"];
        if (this.props.type && ! this.props.active) {
            classes.push("btn-" + this.props.type)
        } else if(this.props.active){
            classes.push("btn-primary");
        }else{
            classes.push("btn-default");
        }
        if (this.props.size) {
            classes.push("btn-" + this.props.size)
        } else {
            classes.push("btn-lg");
        }
        return classes.join(" ");
    }

    render() {
        let text = this.props.text;
        let glyphicon;
        if (this.props.icon) {
            glyphicon = <span className={"glyphicon " + "glyphicon-" + this.props.icon}/>
        }
        return (
            <button
                type="button"
                className={this.getClassName()}
                onClick={this.clickAction.bind(this)}
                disabled={this.props.disabled}>
                {text} {glyphicon}
            </button>
        );
    }

}