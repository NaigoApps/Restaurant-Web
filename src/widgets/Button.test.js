import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import Button from "./Button";

it('basic button', () => {
    const btn = renderer.create(
        <Button/>
    ).toJSON();
    expect(btn).toMatchSnapshot();
});

it('disabled button', () => {
    const btn = renderer.create(
        <Button disabled/>
    ).toJSON();
    expect(btn).toMatchSnapshot();
});

it('text button', () => {
    const btn = renderer.create(
        <Button text="TEST"/>
    ).toJSON();
    expect(btn).toMatchSnapshot();
});

it('icon button', () => {
    const btn = renderer.create(
        <Button icon="ICON"/>
    ).toJSON();
    expect(btn).toMatchSnapshot();
});

it('action button', () => {
    const callback = jest.fn();

    const btn = shallow(<Button commitAction={callback}/>);

    btn.find('button').simulate('click');
    expect(callback.mock.calls.length).toBe(1);
});

// it('action disabled button', () => {
//     const callback = jest.fn();
//
//     const btn = shallow(<Button disabled={true} commitAction={callback}/>);
//
//     btn.find('button').simulate('click');
//     expect(callback.mock.calls.length).toBe(0);
//
// });

