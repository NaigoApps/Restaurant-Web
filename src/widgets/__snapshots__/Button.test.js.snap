// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic button 1`] = `
<button
  className="btn btn-secondary"
  disabled={false}
  onClick={[Function]}
  type="button"
>
   
</button>
`;

exports[`disabled button 1`] = `
<button
  className="btn btn-secondary"
  disabled={true}
  onClick={[Function]}
  type="button"
>
   
</button>
`;

exports[`icon button 1`] = `
<button
  className="btn btn-secondary"
  disabled={false}
  onClick={[Function]}
  type="button"
>
   
  <span
    className="fa fa-ICON"
  />
</button>
`;

exports[`text button 1`] = `
<button
  className="btn btn-secondary"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  TEST
   
</button>
`;
